### DISCUSSION TITLE
How to update MoveStdlib of the account 0x1?

### DISCUSSION THREAD

**Discussion by byron1st** [aptos-labs/aptos-developer-discussions#327] [Questions] [answered]
### Discord user ID

byron1st

### Describe your question in detail.

I'm running several validators privately (not connected to mainnet, so it is operated as a private blockchain). For several months, I updated my validators from the quite old version to the most recent version 1.16.3, steadily. However, today, I found MoveStdlib of the account 0x1 is quite old and there is no new functions added. (For example, my MoveStdlib does not have the function `vector::slice`, which was added at Feb, 2024.)

So, my questions are;

1. Should I update MoveStdlib of the account 0x1 separately regardless of the update of a validator?
2. If then, how can I update MoveStdlib of the account 0x1?
3. Are the AptosFramework and AptosStdlib same with MoveStdlib in terms of their update?

Thanks.

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

Aptos CLI

**Comment by byron1st** [comment]
Ok, I learned more stuff.
I found a guide from https://github.com/aptos-foundation/mainnet-proposals/tree/main?tab=readme-ov-file and I tried to follow this guide. However, whenever I tried to run `aptos governance propose` command, I encountered the below error:

```
Compiling, may take a little while to download git dependencies...
{
  "Error": "Move compilation failed: Unable to resolve packages for package 'SubmitProposal': While resolving dependency 'AptosFramework' in package 'SubmitProposal': While processing dependency 'AptosFramework': Unable to find package manifest for 'AptosFramework' at \"/var/folders/hn/t5_fq8s97z1gkg5gnxbhfqyr0000gp/T/.tmpTIKG0d/./aptos-move/framework/aptos-framework/\""
}
```

I tried this command in aptos-core repo at my local machine. (checkout to aptos-node-v1.16.3 tag)

How can I resolve this error? My `aptos governance propose` command is:

```
aptos governance propose --metadata-url https://my/personal/repo/metadata/v1.16/step_1_upgrade_framework.json \
--pool-address 0x... \
--script-path ./stdlib.move \
--framework-local-dir ./aptos-move/framework/aptos-framework/ \
--is-multi-step \
--profile local
```

Thanks.

**Comment by 0x-j** [comment, marked as answer]
is your framework path correct? i think `Unable to find package manifest` only shows up when there's no move.toml file

**Reply by byron1st** [reply]
Ah, thanks!
My problem is that the path assigned at `--framework-local-dir` is a relative path. I changed it to the absolute path, then it compiles correctly.

(However, I encountered another error, but it is not related to this question. If I cannot resolve this new error, I will open a new discussion then.)

Thanks!
